applications/ExaquteSandboxApplication/custom_processes/divergence_process.cpp

```diff
@@ -0,0 +1,168 @@
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Tosi
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/geometry_utilities.h"
+#include "utilities/variable_utils.h"
+#include "exaqute_sandbox_application_variables.h"
+
+// Application includes
+#include "custom_processes/divergence_process.h"
+
+
+namespace Kratos
+{
+    /* Public functions *******************************************************/
+
+    // Constructor
+    CalculateDivergenceProcess::CalculateDivergenceProcess(
+        ModelPart& rModelPart,
+        Parameters ThisParameters):
+        Process(),
+        mrModelPart(rModelPart)
+    {}
+
+    std::string CalculateDivergenceProcess::Info() const
+    {
+        return "CalculateDivergenceProcess";
+    }
+
+    void CalculateDivergenceProcess::PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream << "CalculateDivergenceProcess";
+    }
+
+    void CalculateDivergenceProcess::PrintData(std::ostream& rOStream) const
+    {
+        this->PrintInfo(rOStream);
+    }
+
+    void CalculateDivergenceProcess::ExecuteInitialize()
+    {
+        KRATOS_TRY;
+
+        auto& r_elements_array = mrModelPart.Elements();
+        // Initialize variables
+        VariableUtils().SetNonHistoricalVariableToZero(DIVERGENCE, r_elements_array);
+        VariableUtils().SetNonHistoricalVariableToZero(VELOCITY_H1_SEMINORM, r_elements_array);
+
+        KRATOS_CATCH("");
+    }
+
+    void CalculateDivergenceProcess::ExecuteBeforeOutputStep()
+    {
+        KRATOS_TRY;
+
+        // Extract information
+        const std::size_t dimension = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+
+        // Check and set number of elements and check dimension
+        KRATOS_ERROR_IF(dimension < 2 || dimension > 3) << "Inconsinstent dimension to execute the process. Dimension =" << dimension << " but needed dimension = 2 or dimension = 3." << std::endl;
+        KRATOS_ERROR_IF(mrModelPart.NumberOfElements() == 0) << "The number of elements in the domain is zero. The process can not be applied."<< std::endl;
+        const unsigned int number_elements = mrModelPart.NumberOfElements();
+
+        // Auxiliar containers
+        GeometryData::ShapeFunctionsGradientsType DN_DX;
+        Vector grad_x;
+        Vector grad_y;
+        Vector grad_z;
+
+        // Iterate over the elements
+        #pragma omp parallel for firstprivate(DN_DX,grad_x,grad_y,grad_z)
+        for(int i_elem = 0; i_elem < static_cast<int>(number_elements); ++i_elem) {
+            auto it_elem = mrModelPart.ElementsBegin() + i_elem;
+            const auto& r_geometry = it_elem->GetGeometry();
+
+            // Current geometry information
+            const std::size_t number_nodes_element = r_geometry.PointsNumber();
+
+            // Build values vectors of the velocity
+            Vector values_x(number_nodes_element);
```

Comment:
Also from @rubenzorrilla suggestions, we can also put the declaration of these variables outside the for loop.